controllers/postulants.js

```diff
@@ -1,201 +1,107 @@
-      family_members: {
-        name: req.query.family_members_name || postulantToUpdate.family_members.name,
-        bond: req.query.family_members_bond || postulantToUpdate.family_members.bond,
-      },
-      availability: {
-        from: req.query.availability_from || postulantToUpdate.availability_from,
-        to: req.query.availability_to || postulantToUpdate.availability_to,
-        day:
-          [
-            req.query.availability_day1 || postulantToUpdate.availability[0],
-            req.query.availability_day2 || postulantToUpdate.availability[1],
-            req.query.availability_day3 || postulantToUpdate.availability[2],
-            req.query.availability_day4 || postulantToUpdate.availability[3],
-            req.query.availability_day5 || postulantToUpdate.availability[4],
-            req.query.availability_day6 || postulantToUpdate.availability[5],
-            req.query.availability_day7 || postulantToUpdate.availability[6],
-          ],
-      },
-    };
-    Postulants[postulantPosition] = postulantToUpdate;
-  } else {
-    res.status(404).send(`Error! Postulant with the ID ${req.params.id} not found. Check and try again.`);
-  }
-  fs.writeFile('./data/postulants.json', JSON.stringify(Postulants), (err) => {
+  Postulants.find({ lastName: req.params.lastName }, (err, result) => {
     if (err) {
-      res.status(500).send('Error while saving the Postulants with the updated Postulant');
+      res.status(500).send(err);
+    }
+    if (!result.length) {
+      res.status(404).send('Postulant not found');
     } else {
-      res.send('Postulant updated!', postulantToUpdate);
+      res.status(200).json(result);
     }
   });
 };
 
-const deletePostulant = (req, res) => {
-  const indexToBeDeleted = Postulants.findIndex((postulant) => postulant.id === parseInt(req.params.id));
-  if (indexToBeDeleted >= 0) {
-    Postulants.splice(indexToBeDeleted, 1);
-    fs.writeFile('./data/postulants.json', JSON.stringify(Postulants), (err) => {
-      if (err) {
-        res.status(500).send('Error while saving the Postulants with the deleted Postulant');
-      } else {
-        res.send('Postulant deleted!');
+const updatePostulant = (req, res) => {
+  Postulants.findByIdAndUpdate(
+    req.params.id,
+    req.body,
+    { new: true },
+    (error, newPostulant) => {
+      if (error) {
+        return res.status(500).json(error);
       }
-    });
-  } else {
-    res.status(404).send('Error. Check the data sent and try again.');
-  }
+      if (!newPostulant) {
+        return res.status(404).json({ msg: `Postulant with the ID ${req.params.id} not found. Check and try again.` });
+      }
+      return res.status(200).json(newPostulant);
+    },
+  );
+};
+
+const deletePostulant = (req, res) => {
+  Postulants.findByIdAndDelete(req.params.id)
+    .then(() => {
+      res.send({
+        msg: `Postulant with id: ${req.params.id} was succesfully deleted.`,
+      });
+    })
```

Comment:
We should check if the found postulant is null and return a 404
```suggestion
  Postulants.findByIdAndDelete(req.params.id)
    .then((res) => {
      if (!res) return res.status(404).json({ msg: `Postulant with id: ${req.params.id} not found.` });
      return res.send({
        msg: `Postulant with id: ${req.params.id} was succesfully deleted.`,
      });
    })
```

Reply:
oh ok! thanks for notice that!